alphagov/paas-docker-cloudfoundry-tools
# [#112819029] Implement automated task generation in Rakefile

https://www.pivotaltracker.com/story/show/112819029

# What

Currently add containers to this repo is tedious and there is lots of repetition. We want to simplify it.

# Context

We refactor the repository to:
 * Automatically generate all the tasks to build and test the containers in the Rakefile based on existing Dockerfiles and spec files.
 * Call the new tasks from travis
 * Refactor the tests to keep all the constants and logic in the same file
 * Update the README to explain how to add containers and remove unnecessary duplication.

# How to test this

Travis builds and runs the tests of the containers, and it should pass for all the containers.

You can also run it manually (docker required):

```
rvm use --create 2.2.2@containers
bundle install
bundle exec rake
```

# who?

Anyone but @keymon
